nosnoc/nosnoc.py

```diff
@@ -659,70 +672,96 @@ def forward_simulation(self, ocp: NosnocOcp, Uk: SX) -> None:
 
         return
 
-    def create_complementarity_constraints(self, sigma_p: SX) -> None:
+
+
+    def create_complementarity(self, x: List[SX], y: SX, sigma: SX) -> None:
+        """
+        adds complementarity constraints corresponding to (x_i, y) for x_i in x to the FiniteElement.
+
+        :param x: list of SX
```

Comment:
I _think_ this should probably take two SX and the summing should be done outside. It's not clear to me why this function does the summation.

Reply:
The idea was that for C functions the sum has to be done more carefully (compared to the bilinear constraint).
And this gives more flexibility.
I discussed this with Armin, but it is kind of ongoing research :D

Reply:
Indeed, the summing should be in general done over the outputs of C-function.
However, as @FreyJo said, it is not clear do we arrive always at an equivalent feasible set with all cross complementarity modes, if we omit lower bounds on the variables entering complementarity conditions. 
If we keep the lower bounds, the equivalence holds but adds no additional value compared to Scholtes variants.